### ISSUE TITLE
Declarations for pkg.phet should be improved

### ISSUE THREAD

**Issue by samreid** [phetsims/chipper#283]
While reviewing #266 I saw code like the following:

```js
  // phetLibs
  pkg.phet = pkg.phet || {};
  pkg.phet.phetLibs = pkg.phet.phetLibs || [];
  pkg.phet.phetLibs.push( pkg.name ); // add the repo that's being built
  pkg.phet.phetLibs = pkg.phet.phetLibs.concat( buildInfo.common.phetLibs ); // add phetLibs from build.json
  if ( global.phet.chipper.brand === 'phet-io' && buildInfo[ 'phet-io' ] && buildInfo[ 'phet-io' ].phetLibs ) {
    pkg.phet.phetLibs = pkg.phet.phetLibs.concat( buildInfo[ 'phet-io' ].phetLibs ); // add phetLibs for phet-io brand
  }
  pkg.phet.phetLibs = _.uniq( pkg.phet.phetLibs.sort() ); // sort and remove duplicates
```

There are two related problems here:
1. The pkg.phet defines required fields such as `requirejsNamespace` and `simTitleStringKey`.  We should not just assign `{}` if pkg. phet is missing, we should throw an assertion if pkg.phet or either of those keys is missing.
2. It could be very confusing to augment/filter pkg.phet.phetLibs.  Why not leave the original alone and come up with a new variable or name for the augmented/filtered phetLibs?  Having this variable deviate from what is in the package.json but still have the same name as the package.json seems misleading and error prone.

@pixelzoom what do you recommend here?

**Comment by samreid** [comment]
Similarly, at the top of Gruntfile.js, we have:

```js
  // for sharing global information
  global.phet = global.phet || {};
  global.phet.chipper = global.phet.chipper || {};
```

In my opinion, it should be an error if either of these was defined before Gruntfile.js started.  Why not either simply assign or assert that is the case?

**Comment by pixelzoom** [comment]
Re phetLibs issue... This is all going away when #232 (buildConfig) is committed, and it's one of the reasons that we created buildConfig.  So ignore it.

Re globals... Proposed change:

```js
// for sharing global information
assert( !global.phet, 'global.phet already exists' );
assert( !global.phet.chipper, 'global.phet.chipper already exists' );
global.phet = {
  chipper: {}
};
```

@samreid Does that look acceptable?

**Comment by pixelzoom** [comment]
Somewhat related to https://github.com/phetsims/chipper/issues/281 (attach all globals to global.het.chipper).

**Comment by samreid** [comment]
How about:

```js
// for sharing global information
assert( !global.phet, 'global.phet already exists' );
global.phet = {
  chipper: {}
};
```

There is no need to check for phet.chipper if we already asserted that phet itself does not exist.

**Comment by pixelzoom** [comment]
Added global.phet verification.
pheLibs cleanup will be addressed by #232.
Closing.
